app/controllers/admin/reports_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Admin::ReportsController < ApplicationController
+
+	def loss_of_pays
+		@loss_of_pays = LossOfPay.includes(:loss_of_pay_infos)
+  end
+
+  def create
+    begin
+      @report = Report.new(params)
+      unless @report.valid?
+        return render action: :loss_of_pays
+      end
+      case params[:report_type].downcase
+        when 'lop'
+          file = @report.generate_lop_report
+          send_file(file)
+        when 'lopRefund'
+          @report.print_lop_refunt
```

Comment:
@ramanbedi1989 - actually i have to change this logic, becz there is no separate lop refund report.
It is combined with lop report.